src/adapters/adapter.js

```diff
@@ -73,10 +103,17 @@ class Adapter {
                 else folders[item.path] = item.children = []
               }
 
-              // encodes but retains the slashes, see #274
-              const encodedPath = path.split('/').map(encodeURIComponent).join('/')
-              item.a_attr = {
-                href: this._getItemHref(repo, type, path)
+              // If item is part of a PR, jump to that file's diff
+              if (item.patch && typeof item.patch.diffId === 'number') {
+                item.a_attr = {
+                  href: `/${repo.username}/${repo.reponame}/pull/${repo.pull}/files#diff-${item.patch.diffId}`
+                }
+              } else {
+                // encodes but retains the slashes, see #274
+                const encodedPath = path.split('/').map(encodeURIComponent).join('/')
```

Comment:
Yes, please. I think you'll need to change the case for patch too, if username/repo has forbidden characters which need encoding.